src/mumble/TalkingUI.cpp

```diff
@@ -395,9 +395,12 @@ TalkingUIUser *TalkingUI::findOrAddUser(const ClientUser *user) {
 		TalkingUIUser *newUserEntry                = userEntry.get();
 
 		// * 1000 as the setting is in seconds whereas the timer expects milliseconds
-		userEntry->setLifeTime(Global::get().s.iTalkingUI_SilentUserLifeTime * 1000);
+		int silentUserLifeTime = Global::get().s.iTalkingUI_SilentUserLifeTime;
+		if (silentUserLifeTime != -1)
+			userEntry->setLifeTime(silentUserLifeTime * 1000);
 
-		userEntry->restrictLifetime(!isSelf || !Global::get().s.bTalkingUI_LocalUserStaysVisible);
+		userEntry->restrictLifetime(silentUserLifeTime != -1
```

Comment:
Do you mean to change the restrictLifetime call to take in two boolean values instead, or just use a temporary variable instead of the `!= -1` check? Or do you mean to go one step further and have a separate global boolean variable?

Reply:
I am thinking of using one or two temporary bools to make the call spanning line 402 and 403 easier to read